.github/actions/yarn-nm-install/action.yml

```
@@ -0,0 +1,88 @@
+########################################################################################
+# "yarn install" composite action for yarn 3/4+ and "nodeLinker: node-modules"         #
+#--------------------------------------------------------------------------------------#
+# Requirement: @setup/node should be run before                                        #
+#                                                                                      #
+# Usage in workflows steps:                                                            #
+#                                                                                      #
+#      - name: 📥 Monorepo install                                                     #
+#        uses: ./.github/actions/yarn-nm-install                                       #
+#        with:                                                                         #
+#          enable-corepack: false # (default)                                          #
+#          cache-install-state: false # (default)                                      #
+#          cache-node-modules: false # (default)                                       #
+#                                                                                      #
+# Reference:                                                                           #
+#   - latest: https://gist.github.com/belgattitude/042f9caf10d029badbde6cf9d43e400a    #
+########################################################################################
+
+name: 'Monorepo install (yarn)'
+description: 'Run yarn install with node_modules linker and cache enabled'
+inputs:
+  enable-corepack:
+    description: 'Enable corepack'
+    required: false
+    default: 'false'
+  cache-node-modules:
+    description: 'Cache node_modules, might speed up link step (invalidated lock/os/node-version/branch)'
+    required: false
+    default: 'false'
+  cache-install-state:
+    description: 'Cache yarn install state, might speed up resolution step when node-modules cache is activated (invalidated lock/os/node-version/branch)'
+    required: false
+    default: 'false'
+
+runs:
+  using: 'composite'
+
+  steps:
+    - name: ⚙️ Enable Corepack
+      if: ${{ inputs.enable-corepack }} == 'true'
+      shell: bash
+      run: corepack enable
+
+    - name: ⚙️ Expose yarn config as "$GITHUB_OUTPUT"
+      id: yarn-config
+      shell: bash
+      env:
+        YARN_ENABLE_GLOBAL_CACHE: "false"
+      run: |
+        echo "CACHE_FOLDER=$(yarn config get cacheFolder)" >> $GITHUB_OUTPUT
+        echo "CURRENT_NODE_VERSION="node-$(node --version | cut -d . -f 1  | sed 's/[^0-9]*//g')".x" >> $GITHUB_OUTPUT
```

```suggestion
        echo "CURRENT_NODE_VERSION="node-$(node --version)"" >> $GITHUB_OUTPUT
```

What do you think about simplifying this a bit? Yes, we would have a few more col-starts over time, but it becomes much more readable and we make sure a new node version also creates a new cache-key.

---

I prefer to keep it inline with the reference gist for now. Runners update versions regularly, it's less the cold cache that this is trying to fix than the cache budget on main branch.

But that's a good point.

---

Cache budget is not a big concern in our case, so I'd prefer simplicity and optimize once we need it here.

---

I've committed the suggestion.... But just an extra reminder - keeping the gist reference for a while helps me to bring easy life to the next iteration (more to come). After that happy to let it live its life :)